modules/gdnative/godot/string.cpp

```diff
@@ -112,6 +113,1149 @@ void GDAPI godot_string_destroy(godot_string *p_self) {
 	self->~String();
 }
 
+/* Standard size stuff */
+
+godot_int GDAPI godot_string_length(const godot_string *p_self) {
+	const String *self = (const String *)p_self;
+
+	return self->length();
+}
+
+/* Helpers */
+
+godot_bool GDAPI godot_string_begins_with(const godot_string *p_self, const godot_string *p_string) {
+	const String *self = (const String *)p_self;
+	const String *string = (const String *)p_string;
+
+	return self->begins_with(*string);
+}
+
+godot_bool GDAPI godot_string_begins_with_char_array(const godot_string *p_self, const char *p_char_array) {
+	const String *self = (const String *)p_self;
+
+	return self->begins_with(p_char_array);
+}
+
+godot_array GDAPI godot_string_bigrams(const godot_string *p_self) {
+	const String *self = (const String *)p_self;
+	Vector<String> return_value = self->bigrams();
+
+	godot_array result;
+	memnew_placement(&result, Array);
+	Array *proxy = (Array *)&result;
+	proxy->resize(return_value.size());
+	for (int i = 0; i < return_value.size(); i++) {
+		(*proxy)[i] = return_value[i];
+	}
+
+	return result;
+};
+
+godot_string GDAPI godot_string_chr(wchar_t p_character) {
+	godot_string result;
+	memnew_placement(&result, String(String::chr(p_character)));
+
+	return result;
+}
+
+godot_bool GDAPI godot_string_ends_with(const godot_string *p_self, const godot_string *p_string) {
+	const String *self = (const String *)p_self;
+	const String *string = (const String *)p_string;
+
+	return self->ends_with(*string);
+}
+
+godot_int GDAPI godot_string_find(const godot_string *p_self, godot_string p_what) {
+	const String *self = (const String *)p_self;
+	String *what = (String *)&p_what;
+
+	return self->find(*what);
+}
+
+godot_int GDAPI godot_string_find_from(const godot_string *p_self, godot_string p_what, godot_int p_from) {
+	const String *self = (const String *)p_self;
+	String *what = (String *)&p_what;
+
+	return self->find(*what, p_from);
+}
+
+godot_int GDAPI godot_string_findmk(const godot_string *p_self, const godot_array *p_keys) {
+	const String *self = (const String *)p_self;
+
+	Vector<String> keys;
+	Array *keys_proxy = (Array *)p_keys;
+	keys.resize(keys_proxy->size());
+	for (int i = 0; i < keys_proxy->size(); i++) {
+		keys.push_back((*keys_proxy)[i]);
```

Comment:
If you `resize()` then it sets the size. If you then `push_back` you increase the size again.

So this will have `keys_proxy->size()` times an empty string at the front, then the real strings in the back.

You should be doing `keys[i] = (*keys_proxy)[i];`, that way you insert them in-place